src/components/ha-related-items.ts

```diff
@@ -0,0 +1,323 @@
+import { HassEntity, UnsubscribeFunc } from "home-assistant-js-websocket";
+import {
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+  CSSResult,
+  css,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import {
+  AreaRegistryEntry,
+  subscribeAreaRegistry,
+} from "../data/area_registry";
+import { ConfigEntry, getConfigEntries } from "../data/config_entries";
+import {
+  DeviceRegistryEntry,
+  subscribeDeviceRegistry,
+} from "../data/device_registry";
+import { SceneEntity } from "../data/scene";
+import { findRelated, ItemType, RelatedResult } from "../data/search";
+import { SubscribeMixin } from "../mixins/subscribe-mixin";
+import { HomeAssistant } from "../types";
+import "./ha-switch";
+
+@customElement("ha-related-items")
+export class HaRelatedItems extends SubscribeMixin(LitElement) {
+  @property() public hass!: HomeAssistant;
+  @property() public itemType!: ItemType;
+  @property() public itemId!: string;
+  @property() private _entries?: ConfigEntry[];
+  @property() private _devices?: DeviceRegistryEntry[];
+  @property() private _areas?: AreaRegistryEntry[];
+  @property() private _related?: RelatedResult;
+
+  public hassSubscribe(): UnsubscribeFunc[] {
+    return [
+      subscribeDeviceRegistry(this.hass.connection!, (devices) => {
+        this._devices = devices;
+      }),
+      subscribeAreaRegistry(this.hass.connection!, (areas) => {
+        this._areas = areas;
+      }),
+    ];
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    getConfigEntries(this.hass).then((configEntries) => {
+      this._entries = configEntries;
+    });
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (
+      (changedProps.has("itemId") || changedProps.has("itemType")) &&
+      this.itemId &&
+      this.itemType
+    ) {
+      this._findRelated();
+    }
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._related) {
+      return html``;
+    }
+    return html`
+      ${this._related.config_entry && this._entries
+        ? this._related.config_entry.map((relatedConfigEntryId) => {
+            const entry: ConfigEntry | undefined = this._entries!.find(
+              (configEntry) => configEntry.entry_id === relatedConfigEntryId
+            );
+            if (!entry) {
+              return;
+            }
+            return html`
+              <h3>
+                ${this.hass.localize(
+                  "ui.components.related-items.integration"
+                )}:
+              </h3>
+              <a
```

Comment:
So not, now, but for a future PR:

We should create web components that render a config entry, device, entity, area etc as linkable entities. Then we can also start adding hover tooltips ala GitHub (hover over my name) in the future.